### ISSUE TITLE
dockerng.sls_build should accept a pillar dictionary (and other parameters from the calling salt environment)

### ISSUE THREAD

**Issue by brianthelion** [saltstack/salt#42202]
### Description of Issue/Question

This is a feature request. With `dockerng.sls_build` [slated to become](https://docs.saltstack.com/en/develop/ref/states/all/salt.states.docker_image.html#salt.states.docker_image.present) `docker_image.present: -sls:`, I think it's worth noting two things:

1. This functionality is really clever and has extremely high utility.
2. Its utility is nonetheless limited by the fact that it's currently impossible to pass anything from the calling salt environment into the called sls.

What kinds of things would one like to pass?

- Pillar and grain updates that have happened during the caller's execution.
- Variables from the local calling scope.

**Comment by gtmanfred** [comment]
This would be great if we could do this.

I am going to mark this as a feature request.

Right now we have some other things that really need to be take care of first before we can do some of these things https://github.com/saltstack/salt/issues/39956

Thanks,
Daniel

**Comment by brianthelion** [comment]
@gtmanfred I looked at [the code](https://github.com/saltstack/salt/blob/d36f0ec0f035f48a0379420e12f0548461f7e033/salt/modules/dockermod.py#L5303) and supporting pillar ingress should be pretty easy. It's simply a matter of updating the interface, and should only take a few edits. Grains are a little more opaque for me.

**Comment by brianthelion** [comment]
@gtmanfred I have pillar ingress working; that was pretty straightforward:
```
my_docker_image:
  docker_image.present:
    - name:
      - foo:latest
    - sls:
      - foo.foo_state
    - base: brianthelion/ubuntu-python:latest
    - pillar:
        target: /tmp
        bar: baz
```

**Comment by brianthelion** [comment]
@gtmanfred See https://github.com/f0cal/salt/commit/1533b6520d701ffb540f6ed2b4d45a5a599278dc

This currently lacks inline documentation updates, but if it looks reasonable I'll take care of that and submit a PR.

**Comment by gtmanfred** [comment]
@brianthelion that seems reasonable, can you submit that PR?

**Comment by jagguli** [comment]
I probably should create another ticket for this but, pillar matching does not work for docker containers,

I can set grains for containers using states, but I cannot compile pillars based on it.

then only option I have is to pass pillar through the state file, thanks to this.
